x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1286,20 +1286,19 @@ public LogicalPlan apply(LogicalPlan plan) {
         private static LogicalPlan normalize(Aggregate aggregate, AttributeMap<Expression> aliases) {
             var aggs = aggregate.aggregates();
             List<NamedExpression> newAggs = new ArrayList<>(aggs.size());
-            boolean changed = false;
```

Comment:
This change is bigger as it moves the replacing the agg/af manually to using transformDown instead.